backend/tests/end_to_end/infrastructures/apps/load/test_api.py

```diff
@@ -0,0 +1,452 @@
+    unauthenticated_client: APIClientData,
+):
+    # Given
+    client = unauthenticated_client.client
+
+    pagination_limit = 1
+    data_1 = model_factories.DataFactory.create()
+    data_2 = model_factories.DataFactory.create()
+
+    # When and then
+    response = client.get(
+        get_url(
+            path_name="load-list",
+            query_params={
+                infrastructure_common_consts.PAGINATION_OFFSET_QUERY_PARAMETER_NAME: 0,
+                infrastructure_common_consts.PAGINATION_LIMIT_QUERY_PARAMETER_NAME: pagination_limit,
+            },
+        ),
+    )
+
+    assert response.status_code == HTTPStatus.OK
+    json_response: dict[str, Any] = response.data
+    assert infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME in json_response
+    assert (
+        UUID(
+            json_response[infrastructure_common_consts.PAGINATION_RESULTS_NAME][0]["id"]
+        )
+        == data_1.id
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_PREVIOUS_LINK_NAME)
+        is None
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME)
+        is not None
+    )
+
+    # When and then
+    response = client.get(
+        json_response[infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME]
+    )
+
+    assert response.status_code == HTTPStatus.OK
+    json_response: dict[str, Any] = response.data  # type: ignore[no-redef]
+    assert infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME in json_response
+    assert (
+        UUID(
+            json_response[infrastructure_common_consts.PAGINATION_RESULTS_NAME][0]["id"]
+        )
+        == data_2.id
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_PREVIOUS_LINK_NAME)
+        is not None
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME)
+        is None
+    )
+
+
+@pytest.mark.parametrize(
+    ("pagination_key", "pagination_value"),
+    (
+        (
+            infrastructure_common_consts.PAGINATION_OFFSET_QUERY_PARAMETER_NAME,
+            "invalid-offset",
+        ),
+        (
+            infrastructure_common_consts.PAGINATION_LIMIT_QUERY_PARAMETER_NAME,
+            "invalid-records_per_page",
+        ),
+    ),
+)
+def test_list_data_endpoint_handles_invalid_pagination_parameters(
+    unauthenticated_client: APIClientData, pagination_key: str, pagination_value: str
+):
+    # Given
+    client = unauthenticated_client.client
+
+    # When
+    response = client.get(
+        get_url(
+            path_name="load-list",
+            query_params={
+                pagination_key: pagination_value,
+            },
+        ),
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.BAD_REQUEST
+    assert infrastructure_common_consts.ERROR_DETAIL_KEY in response.data
+
+
+def test_list_data_endpoint_filtering_by_is_satisfied(
```

Comment:
Can we make this parametrized test? I know it does not make sense, but only to how people.

Remember to adjust the test name.